### PR TITLE
os/bluestore: allocator pruning

### PR DESCRIPTION
**Pull request by ifed01**
Looks like we don't need additional reserve() method in Allocator interface, removing it.

### CODE REVIEW

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -10563,19 +10556,19 @@ int BlueStore::_do_alloc_write(
...
-    return r;
-  }
   PExtentVector prealloc;
   prealloc.reserve(2 * wctx->writes.size());;
   int prealloc_left = 0;
   prealloc_left = alloc->allocate(
     need, min_alloc_size, need,
     0, &prealloc);
+  if (prealloc_left  < 0) {
+    derr << __func__ << " failed to reserve 0x" << std::hex << need << std::dec
```
Comment (liewegas) [inline src/os/bluestore/BlueStore.cc]: s/reserve/allocate/

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2018-05-14_22:14:39-rados-wip-sage3-testing-2018-05-14-1429-distro-basic-smithi/
http://pulpito.ceph.com/sage-2018-05-16_15:59:20-rados-wip-sage3-testing-2018-05-14-1429-distro-basic-smithi/

failures are http://tracker.ceph.com/issues/24150

File: src/os/bluestore/BitMapAllocator.cc [hunk 2]
```diff
@@ -119,14 +90,43 @@ int64_t BitMapAllocator::allocate(
...
      << " want_size " << want_size
      << " alloc_unit " << alloc_unit
      << " hint " << hint
      << dendl;
   hint = hint % m_total_size; // make hint error-tolerant
-  return allocate_dis(want_size, alloc_unit / m_block_size,
+  auto res = allocate_dis(want_size, alloc_unit / m_block_size,
                       max_alloc_size, hint / m_block_size, extents);
+
+  if (res < want_size) {
```
Comment (joscollin) [inline src/os/bluestore/BitMapAllocator.cc]: @ifed01 
This creates a warning now:

```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/os/bluestore/BitMapAllocator.cc: In member function 'virtual int64_t BitMapAllocator::allocate(uint64_t, uint64_t, uint64_t, int64_t, PExtentVector*)':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/os/bluestore/BitMapAllocator.cc:116:11: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (res < want_size) {
       ~~~~^~~~~~~~~~~
```

Comment (ifed01) [reply src/os/bluestore/BitMapAllocator.cc]: @joscollin : thanks for the note, fixed by https://github.com/ceph/ceph/pull/22064

**Pull request merged**